File: smileyFace.js
// You need to install the 'canvas' package first using npm or yarn
// npm install canvas
const { createCanvas } = require('canvas');
const canvas = createCanvas(200, 200);
const ctx = canvas.getContext('2d');

// Draw the face
// ... (same drawing logic as in the browser example)
const canvas = document.getElementById('smileyCanvas');
    const ctx = canvas.getContext('2d');

    // Draw the face
    ctx.beginPath();
    ctx.arc(100, 100, 50, 0, Math.PI * 2, true); // Outer circle
    ctx.fillStyle = 'yellow';
    ctx.fill();
    ctx.lineWidth = 2;
    ctx.stroke();

    // Draw the eyes
    ctx.fillStyle = 'black';
    ctx.beginPath();
    ctx.arc(80, 85, 5, 0, Math.PI * 2, true);  // Left eye
    ctx.arc(120, 85, 5, 0, Math.PI * 2, true);  // Right eye
    ctx.fill();

    // Draw the mouth
    ctx.beginPath();
    ctx.arc(100, 100, 35, 0, Math.PI, false);  // Mouth (clockwise)
    ctx.stroke();

// To display this, you'd need to save it to a file or stream it to a web server
const fs = require('fs');
const out = fs.createWriteStream(__dirname + '/smiley.png');
const stream = canvas.createPNGStream();
stream.pipe(out);
out.on('finish', () => console.log('The PNG file was created.'));

File: SmileyFace.java
import java.awt.Canvas;
import java.awt.Graphics;
import java.awt.Color;
import javax.swing.JFrame;

public class SmileyFace extends Canvas {
    
    public static void main(String[] args) {
        JFrame frame = new JFrame("My Drawing");
        Canvas canvas = new SmileyFace();
        canvas.setSize(200, 200);
        frame.add(canvas);
        frame.pack();
        frame.setVisible(true);
    }

    public void paint(Graphics c) {
        // Draw the outline of the face
        c.setColor(Color.YELLOW);
        c.fillOval(50, 50, 100, 100);  // Change coordinates and size as needed

        // Draw the eyes
        c.setColor(Color.BLACK);
        c.fillOval(85, 75, 10, 10);    // Left Eye
        c.fillOval(105, 75, 10, 10);   // Right Eye

        // Draw the mouth
        c.drawArc(75, 75, 50, 50, 0, -180);  // Smiling mouth
    }
}
